### DISCUSSION TITLE
std::initializer_list adapter

### DISCUSSION THREAD

**Discussion by domcir** [bytedeco/javacpp#626] [Q&A]
Hi
I've checked all presets and in every one std::initializer_list is skipped.
Would it be difficult to expose it?
Thanks
Dom

**Comment by saudet** [comment]
It's not a matter of exposing it, I just don't know how to make it work with pointers. We could kind of consider it as varargs, but if we do that the user wouldn't be able to change the number of elements at runtime.

**Reply by domcir** [reply]
I think it's a good idea but it might not be that simple... you would need to make sure to link std containers exposed as well. I've been experimenting but without any luck so for the time being I just disabled that signature.

**Reply by saudet** [reply]
The problem is that it's not possible to construct a non-empty std::initializer_list without using a hard-coded list:
https://en.cppreference.com/w/cpp/utility/initializer_list/initializer_list
If you find a way to work around this though, please do let me know

**Reply by domcir** [reply]
Agreed, it's a tricky one.
